### DISCUSSION TITLE
Sandboxing - Detection/Prevention of malicious files

### DISCUSSION THREAD

**Discussion by PwXod** [globaleaks/globaleaks-whistleblowing-software#3489] [Q&A]
Good morning Rhyme, Giovanni, sferro,

Imagine that a malicious whistleblower sends us a malicious file (pdf, excel, javascript, payloads, or any type of malicious code) that can compromise the security of our servers.
Does GlobaLeaks include a sandbox to review malicious files uploaded by whistleblowers without compromising our servers?
Where can I document myself on these IDS/IPS, SANDBOX security features implemented in GlobaLeaks?
How does the analysis of malicious files work within GlobaLeaks? What is the security architecture focused on the detection and prevention of malicious files?
Can GlobaLeaks be integrated with Google Recaptcha or whatever to prevent the reporting portal from being easily saturated by robots?

Thank you
Cheers!

**Comment by PwXod** [comment]
Hello,

May I have an answer?

Thank you.

**Comment by evilaliv3** [comment]
Thank you for your question and patience @PwXod 

I actually started writing few weeks ago but i may have not saved the message.

Currently globaleaks do not implement any check on uploads for two reasons:
- we look forward for the possibility to implement end2end encryption and thus the server wont be able to access any data (we are still not there yet)
- there are no good open source technologies that enable to check possible viruses so it is still an hard problem how to possibly implement a sandbox on globaleaks side to test for viruses without involving third parties and possibly violating the confidentiality of the process.
- 
Do you have any suggestion o are you able to support some development on this aspects? 

The following are two references about these aspects:
- https://github.com/globaleaks/GlobaLeaks/issues/711
- https://docs.globaleaks.org/en/main/security/ThreatModel.html#
